File: backend/src/main/java/org/globe42/web/activities/ParticipantDTO.java
package org.globe42.web.activities;

import com.fasterxml.jackson.annotation.JsonUnwrapped;
import org.globe42.domain.Person;
import org.globe42.web.persons.PersonIdentityDTO;

/**
 * Information about a participant to an activity type
 * @author JB Nizet
 */
public final class ParticipantDTO {

    @JsonUnwrapped
    private final PersonIdentityDTO identity;

    private final String email;
    private final String phoneNumber;

    public ParticipantDTO(Person person) {
        this.identity = new PersonIdentityDTO(person);
        this.email = person.getEmail();
        this.phoneNumber = person.getPhoneNumber();
    }

    public PersonIdentityDTO getIdentity() {
        return identity;
    }

    public String getEmail() {
        return email;
    }

    public String getPhoneNumber() {
        return phoneNumber;
    }
}

File: backend/src/main/kotlin/org/globe42/web/activities/ParticipantDTO.kt
package org.globe42.web.activities

import com.fasterxml.jackson.annotation.JsonUnwrapped
import org.globe42.domain.Person
import org.globe42.web.persons.PersonIdentityDTO

/**
 * Information about a participant to an activity type
 * @author JB Nizet
 */
data class ParticipantDTO(@field:JsonUnwrapped val identity: PersonIdentityDTO,
                          val email: String?,
                          val phoneNumber: String?) {

    constructor(person: Person): this(PersonIdentityDTO(person), person.email, person.phoneNumber)
}
